walkeeper/src/replication.rs

```diff
@@ -194,8 +194,14 @@ impl ReplicationConn {
                 break;
             }
         }
-        let (_, timeline) = swh.timeline.get().get_end_of_wal();
-        info!("Start replication from {}", start_pos);
+        let (wal_end, timeline) = swh.timeline.get().get_end_of_wal();
+        let stop_pos: Lsn;
+        if swh.appname == Some("wal_proposer_recovery".to_string()) {
```

Comment:
I'd add comment explaining why this `if` is necessary to prevent deadlock, it will help to not miss changes in the next review.